### PR TITLE
[ci] Run same script to analyze plugins as flutter/plugins.

### LINKED ISSUE #78994: [plugins] [ci] Plugins with custom analysis_options.yaml are failing `analyze` test in flutter/flutter.
The flutter/plugins roller is failing pre-submit:

```
Activated flutter_plugin_tools 0.0.45+3.
▌09:47:03▐ ELAPSED TIME: 5.143s for ../../../k/flutter/bin/cache/dart-sdk/bin/pub global activate flutter_plugin_tools in ../../tmp/t/pluginsDYZTNQ
▌09:47:03▐ RUNNING: cd ../../tmp/t/pluginsDYZTNQ; ../../../k/flutter/bin/cache/dart-sdk/bin/pub global run flutter_plugin_tools analyze
Verifying analysis settings...
Found an extra analysis_options.yaml in /b/s/w/ir/tmp/t/pluginsDYZTNQ/packages/android_alarm_manager/analysis_options.yaml.
If this was deliberate, pass the package to the analyze command with the --custom-analysis flag and try again.
```

https://ci.chromium.org/ui/p/flutter/builders/try/Linux%20flutter_plugins/5356/overview

It seems that the issue was introduced by this change:

https://github.com/flutter/plugins/pull/3703

We need to configure the analysis command to accept custom analysis_options.yaml files, either here:

* https://github.com/flutter/flutter/blob/2e06ed078726e31c5afca36ae9c6736060c7b7e8/dev/bots/test.dart#L851-L860
* Or publish a new version of `flutter_plugin_tools` which allows for custom yaml files.

### PR DESCRIPTION
**Pull request by ditman**
This change runs flutter/plugins tests with the same code that is used in flutter/plugins.

The current method (flutter_plugin_tools) is being deprecated.

Fixes https://github.com/flutter/flutter/issues/78994

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by ditman** [comment]
Publishing PR so I can see the results of the `Linux plugin_test` check.

**Comment by ditman** [comment]
```
▌21:53:28▐ RUNNING: cd ../../tmp/t/pluginsDSSKPB; ../../../k/flutter/script/incremental_build.sh analyze
fatal: No such ref: 'FETCH_HEAD'
fatal: Not a valid object name FETCH_HEAD
Cannot find a merge base for the current branch to run an incremental build...
Please rebase your branch onto the latest master!
```

:/

**Comment by ditman** [comment]
Looking at this one now: https://github.com/flutter/flutter/pull/79008/checks?check_run_id=2188454498

**Comment by ditman** [comment]
The new script succeeded here:

https://ci.chromium.org/ui/p/flutter/builders/try/Linux%20flutter_plugins/b8851841299001934512/overview

File: dev/bots/test.dart [hunk 1]
```diff
@@ -840,23 +840,14 @@ Future<void> _runFlutterPluginsTests() async {
...
-      <String>[
-        'global',
-        'activate',
-        'flutter_plugin_tools',
-      ],
-      workingDirectory: checkout.path,
-    );
-    await runCommand(
-      pub,
+      './script/incremental_build.sh',
```
Comment (stuartmorgan) [inline dev/bots/test.dart:843]: Do we need a `pub get` first, or is that already being explicitly done by an earlier step?

Comment (ditman) [reply dev/bots/test.dart:843]: It seems that incremental_build.sh does a pub get here:

https://github.com/flutter/plugins/blob/master/script/common.sh#L54

**Review by stuartmorgan-g — APPROVED** [review]
LGTM

File: dev/bots/test.dart (hunk 1 above)
Comment (stuartmorgan) [reply dev/bots/test.dart:843]: Oops, I should have known that

**Pull request merged**